pkg/istio/rbac/v1/provider.go

```diff
@@ -96,15 +101,167 @@ func (p *v1) ConvertAthenzModelIntoIstioRbac(m athenz.Model) []model.Config {
+						if err != nil {
+							log.Errorf("error when checking athenz member expiration date, skipping current member: %s, error: %s", roleMember.MemberName, err)
+							continue
+						}
+						if !res {
+							log.Infoln("member expired, skip adding member to authz policy resource, member: ", roleMember.MemberName)
+							continue
+						}
+						res, err = common.CheckAthenzSystemDisabled(roleMember)
+						if err != nil {
+							log.Errorf("error when checking athenz member system disabled, skipping current member: %s, error: %s", roleMember.MemberName, err)
+							continue
+						}
+						if !res {
+							log.Infoln("member expired, skip adding member to authz policy resource, member: ", roleMember.MemberName)
+							continue
+						}
+
+						spiffeName, err := common.MemberToSpiffe(roleMember)
+						if err != nil {
+							log.Errorln("error converting role name to spiffeName: ", err.Error())
+							continue
+						}
+						from_principal.Source.Principals = append(from_principal.Source.Principals, spiffeName)
+						if p.enableOriginJwtSubject {
+							originJwtName, err := common.MemberToOriginJwtSubject(roleMember)
+							if err != nil {
+								log.Errorln(err.Error())
+								continue
+							}
+							from_requestPrincipal.Source.RequestPrincipals = append(from_requestPrincipal.Source.RequestPrincipals, originJwtName)
+						}
+					}
+					//add role spiffee for role certificate
+					roleSpiffeName, err := common.RoleToSpiffe(string(athenzModel.Name), string(roleName))
+					if err != nil {
+						log.Println("error when convert role to spiffe name: ", err.Error())
+						continue
+					}
+					from_principal.Source.Principals = append(from_principal.Source.Principals, roleSpiffeName)
+				}
+			}
+			rule.From = append(rule.From, from_principal)
+			rule.From = append(rule.From, from_requestPrincipal)
+			// form rules_to
+			rule_to := &v1beta1.Rule{}
+			_, err = common.ParseAssertionEffect(assert)
+			if err != nil {
+				log.Debugf(err.Error())
+				continue
+			}
+			method, err := common.ParseAssertionAction(assert)
+			if err != nil {
+				log.Debugf(err.Error())
+				continue
+			}
+			// form rule.To
+			to := &v1beta1.Rule_To{
+				Operation: &v1beta1.Operation{
+					Methods: []string{method},
+				},
+			}
+			if path != "" {
+				to.Operation.Paths = []string{path}
+			}
+			rule_to.To = append(rule_to.To, to)
+			rules = append(rules, rule_to)
+			rules = append(rules, rule)
+		}
+	}
+	spec.Rules = rules
+	out.Spec = spec
+	return out
+}
+
+// GetCurrentIstioRbac returns the authorization policies resources for the specified model's namespace
+// if serviceName is "", return the all the authorization policies in the given namespace,
+// if serviceName is specific, return single authorization policy matching with serviceName.
+func (p *v1) GetCurrentIstioAuthzPolicy(m athenz.Model, csc model.ConfigStoreCache, serviceName string) []model.Config {
```

Comment:
Could we use the GetCurrentIstioRbac function from the interface? We could update it to pass in the service name.